api/services/v1alpha1/gateway_types.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/api/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	GatewayServiceName = "gateway"
+	// GatewayInstanceName the name of the Gateway instance singleton.
+	// value should match whats set in the XValidation below
+	GatewayInstanceName = "default-gateway"
+	GatewayKind         = "Gateway"
+)
+
+// Check that the component implements common.PlatformObject.
+var _ common.PlatformObject = (*Gateway)(nil)
+
+// GatewaySpec defines the desired state of Gateway
+type GatewaySpec struct {
+	// Authentication configuration
+	// +optional
+	Auth GatewayAuthSpec `json:"auth"`
+
+	// Certificate management
+	// +optional
+	Certificates GatewayCertSpec `json:"certificates"`
+
+	// Domain configuration for the gateway
+	// +optional
+	Domain string `json:"domain,omitempty"`
+
+	// Namespace where the gateway resources should be deployed
+	// +kubebuilder:default="openshift-ingress"
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// GatewayAuthSpec defines authentication configuration for the gateway
+type GatewayAuthSpec struct {
+	// Authentication mode: "openshift-oauth" | "oidc" | "auto"
+	// +kubebuilder:validation:Enum=openshift-oauth;oidc;auto
+	// +kubebuilder:default="auto"
+	// +optional
+	Mode string `json:"mode,omitempty"`
+
+	// OIDC configuration (required when mode="oidc")
+	// +optional
```

Comment:
```suggestion
	// OIDC configuration (required when mode="oidc")
	// +kubebuilder:validation:XValidation:rule="self.mode != 'oidc' || has(self.oidc)",message="OIDC configuration is required when mode is 'oidc'"
	// +optional
```

optional suggestion, not sure this is important.